src/platform/sdl/main.c

```diff
@@ -321,3 +337,64 @@ static void mSDLDeinit(struct mSDLRenderer* renderer) {
 
 	SDL_Quit();
 }
+
+void setLogger(struct mCore* core) {
+	int fakeBool = 0;
+	bool logToStdout = true;
+	bool logToFile = false;
+
+	if (mCoreConfigGetIntValue(&core->config, "logToStdout", &fakeBool))
+		logToStdout = fakeBool;
+	if (mCoreConfigGetIntValue(&core->config, "logToFile", &fakeBool))
+		logToFile = fakeBool;
+	const char* logFile = mCoreConfigGetValue(&core->config, "logFile");
+	int filterLevels = core->opts.logLevel;
+
+	_setLogger(logToStdout, logToFile, logFile, filterLevels);
+}
+
+void _setLogger(bool logToStdout, bool logToFile, const char* logFile, int filterLevels) {
+	// Assign basic static variables
+	_logToStdout = logToStdout;
+	_logFile = NULL;
+	
+	if(logToFile && logFile)
+		_logFile = VFileOpen(logFile, O_WRONLY | O_CREAT | O_TRUNC);
+
+	// Create the filter
+	mLogFilterInit(&_filter);
+	mLogFilterSet(&_filter, "gba.bios", mLOG_STUB | mLOG_FATAL);
+	mLogFilterSet(&_filter, "core.status", mLOG_ALL & ~mLOG_DEBUG);
+	_filter.defaultLevels = filterLevels;
+
+	// Fill the logger
+	_logger.log = _mCoreLog;
+	_logger.filter = &_filter;
+}
```

Comment:
Why are these separate functions if they're only ever called once?

Reply:
I wanted to keep them logically separated, but you're right, as just keeping them one after the other should suffice.

Reply:
Fixed in https://github.com/mgba-emu/mgba/pull/2345/commits/9440ca24e9b66396445955640548bab981801c2d .